core/storage/base_model/gae_models.py

```diff
@@ -432,6 +432,112 @@ def _fetch_page_sorted_by_last_updated(
             result[2])
 
 
+class BaseHumanMaintainedModel(BaseModel):
+    """A model that tracks the last time it was updated by a human.
+
+    When the model is updated by a human, use `self.put_for_human()` to store
+    it. For example: after the title of an exploration is changed
+    by its creator.
+
+    Otherwise, when the model is updated by some non-human or automated process,
+    use `self.put_for_bot()`. For example: by a one-off job updating the
+    schema version of a property.
+    """
+
+    # When this entity was last updated on behalf of a human.
+    last_updated_by_human = (
+        datastore_services.DateTimeProperty(indexed=True, required=True))
+
+    def put(self):
+        """Unsupported operation on human-maintained models."""
+        raise NotImplementedError('Use put_for_human or put_for_bot instead')
+
+    def put_for_human(self):
+        """Stores the model instance on behalf of a human."""
+        self.last_updated_by_human = datetime.datetime.utcnow()
+        return super(BaseHumanMaintainedModel, self).put()
+
+    def put_for_bot(self):
+        """Stores the model instance on behalf of a non-human."""
+        return super(BaseHumanMaintainedModel, self).put()
+
+    def put_async_for_human(self):
+        """Stores the model instance asynchronously on behalf of a human."""
+        self.last_updated_by_human = datetime.datetime.utcnow()
+        return super(BaseHumanMaintainedModel, self).put_async()
+
+    def put_async_for_bot(self):
+        """Stores the model instance asynchronously on behalf of a non-human."""
+        return super(BaseHumanMaintainedModel, self).put_async()
```

Comment:
I think it would be fine to not implement these, since we are not sure how realiable the async methods are. Same for the multi async methods.

Reply:
@vojtechjelinek Why do you think put_async and put_multi_async are unreliable? Async methods are usually advantageous right.

Reply:
Okay, the reliability is a guess since we saw a lot of missing commit log and snapshot models (that were created by `put_async`). I would be for forbidding async puts for now and introduce them when we need optimizations and we can properly test it.

Reply:
I have added a lint check for put_multi_async also. Done.